### ISSUE TITLE
Cannot import datadog_dashboard with Event Stream widget

### ISSUE THREAD

**Issue by i-volkov** [DataDog/terraform-provider-datadog#1753]
### Terraform Version

```
Terraform v1.3.7
on darwin_amd64
+ provider registry.terraform.io/datadog/datadog v3.20.0
```

### Affected Resource(s)

-   datadog_dashboard

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    datadog = {
      source  = "DataDog/datadog"
      version = "~> 3.20.0"
    }
  }

  required_version = ">= 1.0.0"
}

variable "datadog_api_key" {
  description = "Do not explicitly set the value, as it will be automatically injected by the TFC workspace."
}
variable "datadog_app_key" {
  description = "Do not explicitly set the value, as it will be automatically injected by the TFC workspace."
}

provider "datadog" {
  api_key = var.datadog_api_key
  app_key = var.datadog_app_key
}

resource "datadog_dashboard" "event-stream-reproduce" {
  title = "Event Stream Reproduce"
  layout_type = "ordered"

  widget {
    event_stream_definition {
      event_size  = "l"
      query       = "ops.group.metric"
      title       = "Widget title"
      title_align = "left"
      title_size  = "16"
    }
  }
}
```

### Debug Output

https://gist.github.com/i-volkov/4eae88dd3dbd359217f426e5879c76a8

### Expected Behavior

Dashboard should be successfully imported

### Actual Behavior

Import command fails with:
`Error: object contains unparsed element: map[columns:[] query:map[data_source:event_stream event_size:l query_string:ops.group.metric] response_format:event_list]`

### Steps to Reproduce

`terraform import datadog_dashboard.event-stream-reproduce $dashboard_id`

### Steps to Reproduce

This issue started to occur a week ago, previously such kind of imports were working fine.

**Comment by nkzou** [comment]
Hi there, thanks for reporting this issue. A fix has been merged and a new release containing it will be published within a few days.

To futureproof, we recommend migrating from the `event_stream` definition to the `list_stream` definition, using `data_source = event_stream`, as `event_stream` will eventually be deprecated.

Here's an example to help migrate - the two widgets in this dashboard are roughly equivalent
```
resource "datadog_dashboard" "event-stream-examples" {
  title = "Event Stream Examples"
  layout_type = "ordered"

  widget {
    event_stream_definition {
      event_size  = "l"
      query       = ""
      title       = "Widget title"
      title_align = "left"
      title_size  = "16"
    }
  }
  widget {
    list_stream_definition {
      title       = "Widget title"
      title_align = "left"
      title_size  = "16"

      request {
        response_format = "event_list"
        columns {
          field = "source"
          width = "auto"
        }
        query {
          data_source  = "event_stream"
          event_size   = "l"
          query_string = ""
        }
      }
    }
  }
}
```

**Comment by i-volkov** [comment]
Thank you for the prompt fix!

**Comment by francovp** [comment]
@nkzou Hello, I got this error when using that example:

```
Error: Unsupported argument
│ 
│   on ../datadog.tf line 75, in resource "datadog_dashboard" "four_golden_signals_dashboard":
│   75:                 event_size   = "l"
│ 
│ An argument named "event_size" is not expected here.
```

I'm using Datadog provider version 3.20.0

Any idea how to solve it?

**Comment by francovp** [comment]
I delete the previous `event_size` argument, and now I'm getting a different error:

```
`invalid value 'event_stream' for ListStreamSource: valid values are
│ [logs_stream audit_stream rum_issue_stream apm_issue_stream
│ logs_pattern_stream]`
```

[But the docs says that `event_stream` it's supported](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/dashboard#nestedblock--widget--list_stream_definition--request--query). What I'm missing?

**Comment by nkzou** [comment]
Hi, sorry for the delay. We released a new version of the provider v3.21.0 that contains the fixes just now, could you try again with that version? Thanks.

**Comment by francovp** [comment]
@nkzou Thanks for the update, using v3.21.0 unfortunately I receive this API error:

```sh
Error: error creating dashboard from https://api.datadoghq.com/api/v1/dashboard: 400 Bad Request: 
{"errors":["Invalid widget definition at position 0 of type group. 
Error: Invalid inner widget definition at position 2 of type list_stream. 
Error: {u'query_string': u'cluster_name:my-cluster-name', u'data_source': u'event_stream', u'event_size': u''} 
is not valid under any of the given schemas.
> - invalid_inner_widget=<{\"definition\": {\"type\": \"list_stream\", \"requests\": [{\"query\": 
{\"query_string\": \"cluster_name:my-cluster-name\", \"data_source\": \"event_stream\", \"event_size\": \"\"},
 \"response_format\": \"event_list\", \"columns\": 
[{\"field\": \"source\", \"width\": \"auto\"}]}], 
\"title_align\": \"left\", \"title_size\": \"16\", \"title\": \"Clusters Events - my-cluster-name\"}}."]}
````

This is using the following code:

```hcl
resource "datadog_dashboard" "four_golden_signals_dashboard" {
  title            = var.title
  description      = "This dashboard was created by using the Datadog provider in Terraform"
  layout_type      = var.layout_type
  restricted_roles = var.restricted_roles

  widget {
    group_definition {
      title            = "General"
      layout_type      = "ordered"
      background_color = "vivid_purple"

      dynamic "widget" {
        for_each = toset(var.clusters[*].name)
        content {
          list_stream_definition {
            title       = "Clusters Events - ${widget.value}"
            title_align = "left"
            title_size  = "16"

            request {
              response_format = "event_list"
              columns {
                field = "source"
                width = "auto"
              }
              query {
                data_source  = "event_stream"
                query_string = "cluster_name:${widget.value}"
              }
            }
          }
        }
      }
    }
  }
}
```

Any ideas what could be the problem?

Thanks in advance
